app/pages/sections/supplier/routes.js

```diff
@@ -0,0 +1,79 @@
+import express from 'express';
+import { ErrorContext } from 'buying-catalogue-library';
+import { logger } from '../../../logger';
+import config from '../../../config';
+import { withCatch, extractAccessToken } from '../../../helpers/routerHelper';
+import {
+  getSupplierSearchPageContext,
+  validateSupplierSearchForm,
+  getSupplierSearchPageErrorContext,
+  findSuppliers,
+} from './search/controller';
+import { getSupplierSelectPageContext } from './select/controller';
+
+const router = express.Router({ mergeParams: true });
+
+export const supplierRoutes = (authProvider, addContext, sessionManager) => {
+  router.get('/', authProvider.authorise({ claim: 'ordering' }), withCatch(authProvider, async (req, res) => {
+    const { orderId } = req.params;
+    logger.info('redirecting to suppliers search page');
+    res.redirect(`${config.baseUrl}/organisation/${orderId}/supplier/search`);
+  }));
+
+  router.get('/search', authProvider.authorise({ claim: 'ordering' }), withCatch(authProvider, async (req, res) => {
+    const { orderId } = req.params;
+    const context = await getSupplierSearchPageContext({ orderId });
+    logger.info(`navigating to order ${orderId} suppliers search page`);
+    res.render('pages/sections/supplier/search/template.njk', addContext({ context, user: req.user, csrfToken: req.csrfToken() }));
+  }));
+
+  router.post('/search', authProvider.authorise({ claim: 'ordering' }), withCatch(authProvider, async (req, res) => {
+    const { orderId } = req.params;
+
+    const response = validateSupplierSearchForm({ data: req.body });
+
+    if (response.success) {
+      const accessToken = extractAccessToken({ req, tokenType: 'access' });
+
+      const suppliersFound = await findSuppliers({
+        name: req.body.supplierName, accessToken,
+      });
+
+      if (suppliersFound.length > 0) {
+        sessionManager.saveToSession({ req, key: 'suppliersFound', value: suppliersFound });
```

Comment:
are these async functions?

Reply:
No. Doesn't have to make a call. Just gets the values out of the request.